include/swift/AST/DiagnosticsSema.def

```diff
@@ -3683,6 +3681,12 @@ NOTE(descriptive_generic_type_declared_here,none,
 
 ERROR(placeholder_type_not_allowed,none,
       "type placeholder not allowed here", ())
+ERROR(placeholder_type_not_allowed_in_return_type,none,
+      "type placeholder may not appear in function return type", ())
+ERROR(placeholder_type_not_allowed_in_parameter,none,
+      "type placeholder may not appear in top-level parameter", ())
+NOTE(replace_placeholder_with_inferred_type,none,
+     "replace the placeholder with the correct type %0", (Type))
```

Comment:
Both `the` are extraneous here according to Diagnostics.md and I'm not sure whether we should use `correct` at all...